lib/spack/env/cc

```diff
@@ -385,52 +383,30 @@ case "$mode" in
         flags=("${flags[@]}" "${SPACK_LDFLAGS[@]}") ;;
 esac
 
+# Prepend include directories
+IFS=':' read -ra include_dirs <<< "$SPACK_INCLUDE_DIRS"
+if [[ $mode == cpp || $mode == cc || $mode == as || $mode == ccld ]]; then
+    for include_dir in "${include_dirs[@]}"; do
+        includes=("${includes[@]}" "$include_dir")
+    done
+fi
 
-# Include the package's prefix/lib[64] dirs in rpath. We don't know until
```

Comment:
Can you add a test that covers both the lib/include directory generation and the cc wrapper, so that we guarantee that this hasn't changed any of the ordering of the commands emitted by the wrapper?

Reply:
To make sure I understand: I assume you mean ordering of options like `-I`/`-L`/`-rpath`

Reply:
Yes. Editing my comment for clarity for future readers.

Reply:
For ordering `-I`, `-L`, and `-rpath` WRT one another, the existing `cc` wrapper tests still enforce that (e.g. that `-I` entries appear before `-L` entries). For ordering within the group of `-L` arguments, that should be addressed by the test added for https://github.com/spack/spack/pull/8136#discussion_r256120589.